symbolicationWebService.py

```diff
@@ -68,10 +68,60 @@ def items(self, section, raw=False, vars=None):
               lambda item: item[0] not in defaults,
               self.superClass.items(self, section, raw, vars))
 
-class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
-  pass
+def initializeSubprocess(options):
+  global gSymFileManager
+
+  SetLoggingOptions(options)
+
+  # Create the .SYM cache manager singleton
+  gSymFileManager = SymFileManager(options)
+
+  # Prefetch recent symbols
+  gSymFileManager.PrefetchRecentSymbolFiles()
+
 
-class RequestHandler(BaseHTTPRequestHandler):
+def processSymbolicationRequest(rawRequest):
+  try:
+    decodedRequest = json.loads(rawRequest)
+    request = SymbolicationRequest(gSymFileManager, decodedRequest)
+    if not request.isValidRequest:
+      LogDebug("Unable to parse request")
+      return None
+
+    response = { 'symbolicatedStacks': [] }
+    for stackIndex in range(len(request.stacks)):
+      symbolicatedStack = request.Symbolicate(stackIndex)
+
+      # Free up memory ASAP
+      request.stacks[stackIndex] = []
+
+      response['symbolicatedStacks'].append(symbolicatedStack)
+
+    response['knownModules'] = request.knownModules[:]
+    if not request.includeKnownModulesInResponse:
+      response = response['symbolicatedStacks']
+
+    request.Reset()
+
+    return json.dumps(response)
+  except Exception as exc:
+    LogDebug("Unable to parse request body: " + str(exc))
+    return None
+
+class DebugHandler(RequestHandler):
+  def get(self, path):
+    self.post(path)
+
+  def post(self, path):
+    # Check the real ip header for the case the remote is behind a proxy
+    xRealIP = self.request.headers.get("X-Real-IP")
+    clientIP = self.request.remote_ip if not xRealIP else xRealIP
```

Comment:
wouldn't this allow debugging from IPs other than the host on which the server is running?